Core/include/Acts/Surfaces/SurfaceArray.hpp

```diff
@@ -203,74 +134,80 @@ class SurfaceArray {
     /// @param surfaces Input surface pointers
     void fill(const GeometryContext& gctx,
               const SurfaceVector& surfaces) override {
-      for (const auto& srf : surfaces) {
-        Vector3 pos = srf->referencePosition(gctx, AxisDirection::AxisR);
-        lookup(pos).push_back(srf);
-      }
+      /// map surface center to grid
+      const auto surfaceToBinMapping = [&](const Surface& surface) {
```

Comment:
Consider breaking this out into the function, if the capture doesn't pull in **too many** things. Otherwise disregard.